runtime/src/main/java/org/corfudb/runtime/LogReplicationClient.java

```diff
@@ -0,0 +1,277 @@
+import com.google.protobuf.Timestamp;
+import org.corfudb.runtime.LogReplication.ClientInfoKey;
+import org.corfudb.runtime.LogReplication.SinksInfoVal;
+import org.corfudb.runtime.LogReplication.LRClientId;
+import org.corfudb.runtime.LogReplication.LRClientInfo;
+import org.corfudb.runtime.collections.CorfuStore;
+import org.corfudb.runtime.collections.Table;
+import org.corfudb.runtime.collections.TableOptions;
+import org.corfudb.runtime.collections.TxnContext;
+import org.corfudb.runtime.exceptions.unrecoverable.UnrecoverableCorfuInterruptedError;
+import org.corfudb.util.retry.ExponentialBackoffRetry;
+import org.corfudb.util.retry.IRetry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.InvocationTargetException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
+
+import static org.corfudb.runtime.view.TableRegistry.CORFU_SYSTEM_NAMESPACE;
+
+/**
+ * A client for log replication
+ */
+public class LogReplicationClient {
+
+    public static final String LR_REGISTRY_TABLE_NAME = "LogReplicationRegistrationTable";
+    public static final String LR_SOURCE_METADATA_TABLE_NAME = "LogReplicationSourceMetadataTable";
+
+    /**
+     * Wait interval between consecutive sync attempts to cap exponential back-off
+     */
+    private static final int SYNC_THRESHOLD = 120;
+
+    private final CorfuStore corfuStore;
+    private final Table<LRClientId, LRClientInfo, Message> registryTable;
+    private final Table<ClientInfoKey, SinksInfoVal, Message> sourceMetadataTable;
+
+    private final LRClientId clientKey;
+    private final LRClientInfo clientInfo;
+
+    private final String clientName;
+    private final String modelEnum;
+
+    private final Logger log;
+
+    /**
+     * Constructor for LogReplicationClient
+     *
+     * @param runtime Corfu Runtime
+     * @param clientName String representation of the client name
+     * @throws InvocationTargetException InvocationTargetException
+     * @throws NoSuchMethodException NoSuchMethodException
+     * @throws IllegalAccessException IllegalAccessException
+     */
+    public LogReplicationClient(CorfuRuntime runtime, String clientName) throws InvocationTargetException, NoSuchMethodException, IllegalAccessException {
+        this.corfuStore = new CorfuStore(runtime);
+        this.clientName = clientName;
+
+        this.modelEnum = "LOGICAL_GROUPS";
+
+        this.clientKey = LRClientId.newBuilder()
+                .setClientName(clientName)
+                .build();
+        this.clientInfo = LRClientInfo.newBuilder()
+                .setClientName(clientName)
+                .setModel(LogReplication.ReplicationModel.valueOf(modelEnum))
+                .setRegistrationTime(Timestamp.newBuilder().setSeconds(new Date().getTime() / 1000).build())
+                .build();
+
+        this.log = LoggerFactory.getLogger("lr-client");
```

Comment:
By reading the code and comments, it looks like there are multiple LR clients. so, do we need different loggers for these different clients? Here the logger name is generic (lr-client)